### ISSUE TITLE
[Chapter 17] - Autoencoder missing Dropout or Gaussian Noise

### ISSUE THREAD

**Issue by dleacock** [ageron/handson-ml2#122]
For Question #9 - Should this model include a Dropout or Gaussian Noise layer to be considered a denoising autoencoder? Between Flatten and the first Dense layer?

```
pretrained_clf = keras.models.Sequential([
    conv_encoder_clone,
    keras.layers.Flatten(),
    keras.layers.Dense(20, activation="selu"),
    keras.layers.Dense(10, activation="softmax")
])
```

Also the solutions for Chapter 17 include the solution to the Semantic Hashing problem but this is a question in the 1st edition and not the 2nd.

**Comment by ageron** [comment]
Hi @dleacock ,
Thanks for your question. You want to include a dropout layer or a Gaussian noise layer (whichever you prefer) directly as the first layer in your model, and the model must output an image. The labels during training should just be the input images. Once you are satisfied with your model, you can reuse the first few layers (excluding the dropout or Gaussian noise layer) to create a classification model. Since you are reusing the pretrained model, you should freeze the reused layers for the first few training epochs, then unfreeze them and continue training with a lower learning rate.
I'm currently working on the solutions to the coding exercises in Chapter 16, and I'll work on Chapter 17 just after that.
Hope this helps!
